src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,128 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WP;
+
+/**
+ * Class AssetSuggestionsService
+ *
+ * Suggest assets and possible final URLs.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Ads
+ */
+class AssetSuggestionsService implements Service {
+
+	/**
+	 * AssetSuggestionsService constructor.
+	 *
+	 * @param WP $wp
+	 */
+	public function __construct( WP $wp ) {
+		$this->wp = $wp;
+	}
+
+	/**
+	 * Get posts that can be used to suggest assets
+	 *
+	 * @return array
+	 */
+	protected function get_post_suggestions(): array {
+		$post_suggestions    = [];
+		$excluded_post_types = [ 'attachment' ];
+
+		$post_types = $this->wp->get_post_types(
+			[
+				'exclude_from_search' => false,
+				'public'              => true,
+			]
+		);
+
+		// Exclude attachment post_type
+		$filtered_post_types = array_diff( $post_types, $excluded_post_types );
+
+		$args = [
+			'post_type'      => $filtered_post_types,
+			'posts_per_page' => -1,
+			'post_status'    => 'publish',
+		];
+
+		$posts = $this->wp->get_posts( $args );
+
+		foreach ( $posts as $post ) {
+			$post_suggestions[] = $this->format_page_response( $post->ID, 'post', $post->post_type, $post->post_title, get_permalink( $post->ID ) );
+		}
+
+		return $post_suggestions;
+	}
+
+	/**
+	 * Get terms that can be used to suggest assets
+	 *
+	 * @return array
+	 */
+	protected function get_terms_suggestion(): array {
+		$terms_suggestions = [];
+
+		// Get all taxonomies that are public, show_in_menu = true helps to exclude taxonomies such as "product_shipping_class".
+		$taxonomies = $this->wp->get_taxonomies(
+			[
+				'public'       => true,
+				'show_in_menu' => true,
+			],
+		);
+
+		// Skip empty terms
+		$terms = $this->wp->get_terms(
+			[
+				'taxonomy'   => $taxonomies,
+				'hide_empty' => true,
```

Comment:
I guess this is true by default. 

Btw. Why we don't allow to show empty terms?

Reply:
> I guess this is true by default.

Removed https://github.com/woocommerce/google-listings-and-ads/pull/1784/commits/77c3744533071f3f008bc91408e9d245457c36e4

> Btw. Why we don't allow to show empty terms?

Because if it does not contain any product/page/post etc, the assets that we can populate like images are limited and I am not sure if the user would like to promote a category/term with no items.

Reply:
Ok, I understand the logic. I was thinking in the case of a user want to fill the campaigns for the terms and after that adding products, not sure if this is a real use case tho.

Reply:
I have changed it to 'hide_empty' => false and it will be up to the user if they want to promote the page or not.